tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/AwsSnsCommon.cs

```diff
@@ -0,0 +1,60 @@
+// <copyright file="AwsSnsCommon.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS
+{
+    internal static class AwsSnsCommon
+    {
+        private const string DatadogAwsSnsServiceName = "aws-sns";
+        private const string SnsOperationName = "sns.request";
+        private const string SnsServiceName = "SNS";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(AwsSnsCommon));
+
+        internal const string IntegrationName = nameof(Configuration.IntegrationId.AwsSns);
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.AwsSns;
+
+        public static Scope CreateScope(Tracer tracer, string operation, out AwsSnsTags tags, ISpanContext parentContext = null)
```

Comment:
If you add `#nullable enable`, you'll need to update various bits to be more declarative

```suggestion
        public static Scope? CreateScope(Tracer tracer, string operation, out AwsSnsTags? tags, ISpanContext? parentContext = null)
```